### PR TITLE
Move listener config from 'cache' block to top-level 'listener' blocks.  Allow cache without auto-auth.

### PR DESCRIPTION
**Pull request by ncabatoff**
Doing this since we want to use the Agent listener for #6384.

### CODE REVIEW

File: command/agent/config/config.go [hunk 1]
```diff
@@ -112,11 +112,26 @@ func LoadConfig(path string, logger log.Logger) (*Config, error) {
...
+	if err != nil {
+		return nil, errwrap.Wrapf("error parsing 'listeners': {{err}}", err)
+	}
+
 	err = parseCache(&result, list)
 	if err != nil {
 		return nil, errwrap.Wrapf("error parsing 'cache':{{err}}", err)
 	}
 
+	if result.Cache != nil {
```
Comment (calvn) [inline command/agent/config/config.go]: Maybe move this logic into `parseCache`

Comment (ncabatoff) [reply command/agent/config/config.go]: I thought about it, the reason I was reluctant is that then parseCache assumes a particular ordering of the calls in LoadConfig.

Comment (calvn) [reply command/agent/config/config.go]: I see, we can leave it as-is then.

File: command/agent.go [hunk 2]
```diff
@@ -202,10 +202,13 @@ func (c *AgentCommand) Run(args []string) int {
...
 		return 1
 	}
-	if config.AutoAuth == nil {
-		c.UI.Error("No auto_auth block found in config file")
+	if config.AutoAuth == nil && config.Cache == nil {
+		c.UI.Error("No auto_auth or cache block found in config file")
 		return 1
 	}
+	if config.AutoAuth == nil {
+		c.UI.Info("No auto_auth block found in config file, not starting auto_auth")
```
Comment (jefferai) [inline command/agent.go]:
```suggestion
		c.UI.Info("No auto_auth block found in config file, not starting automatic authentication feature")
```

**Pull request merged**